up42/viztools.py

```diff
@@ -308,29 +322,42 @@ def map_results(
                 "You first need to download the results via job.download_results()!"
             )
 
-        # Add features to map.
-        # Some blocks store vector results in an additional geojson file.
-        # pylint: disable=not-an-iterable
-        json_fp = [fp for fp in self.results if fp.endswith(".geojson")]
-        if json_fp:
-            json_fp = json_fp[0]
-        else:
+        f_paths = []
+        if isinstance(self.results, list):
+            # Add features to map.
+            # Some blocks store vector results in an additional geojson file.
             # pylint: disable=not-an-iterable
-            json_fp = [fp for fp in self.results if fp.endswith(".json")][0]
+            json_fp = [fp for fp in self.results if fp.endswith(".geojson")]
```

Comment:
Can this and the next lines be simplified into one list comprehension? E.g. by using "if os.path.splitext(fp)[0] in ["json", "geojson"] or similar.

Reply:
I agreed. I would wait for @chrieke and understand the context before engaging in a refactor rampage! :D

Reply:
see comment above, the geojson has higher priority, so needs to be handled separatly.